task_reactredux/task1/src/reducers/index.js

```diff
@@ -0,0 +1,60 @@
+import { validateMail, validatePassword } from '../Validation';
+
+const initialState = {
+    mail: '',
+    password: '',
+    mailValid: false,
+    passwordValid: false,
+    wasSubmited: false
+}
+
+const LoginReduxReducer = (state = initialState, action) => {
```

Comment:
You could split this big reducer into 3: `EmailReducer`, `PasswordReducer`, `FormReducer` and then use `combineReducers` (https://redux.js.org/api/combinereducers)